RetrospectiveExtension.Frontend/components/__tests__/feedbackBoardContainer.test.tsx

```diff
@@ -1,15 +1,62 @@
-import { FeedbackBoardContainerProps } from '../feedbackBoardContainer';
-import { mockWorkflowStage } from '../__mocks__/WorkflowStage';
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { mocked } from 'jest-mock';
+import FeedbackBoardContainer, { FeedbackBoardContainerProps } from '../feedbackBoardContainer';
 
-jest.mock('../workflowStage', () => { return mockWorkflowStage; });
+const getTeamIterationsMock = () => {
+  return [
+    mocked({
+      attributes: mocked({
+        finishDate: new Date(),
+        startDate: new Date(),
+        timeFrame: 1,
+      }),
+      id: 'iterationId',
+      name: 'iteration name',
+      path: 'default path',
+      _links: [],
+      url: 'https://teamfieldvaluesurl'
+    })
+  ];
+};
+
+const getTeamFieldValuesMock = () => {
+  return [
+    mocked({
+      defaultValue: 'default field value',
+      field: mocked({
+        referenceName: 'default reference name',
+        url: 'https://fieldurl'
+      }),
+      values: [
+        mocked({
+          includeChildren: false,
+          value: 'default team field value',
+        })
+      ],
+      links: [],
+      url: 'https://teamfieldvaluesurl'
+    })]
+};
+
+jest.mock('../feedbackBoardMetadataForm', () => { return mocked({});});
+jest.mock('azure-devops-extension-api/Work/WorkClient', () => {
+  return {
+    getTeamIterations: getTeamIterationsMock,
+    getTeamFieldValues: getTeamFieldValuesMock,
+  };
+});
 
 const feedbackBoardContainerProps: FeedbackBoardContainerProps = {
   isHostedAzureDevOps: false,
-  projectId: "1",
+  projectId: '1',
 };
 
-describe(`The Feedback Board Component`, () => {
-  it("Renders a Feedback Board Container Component", () => {
-    console.log(feedbackBoardContainerProps);
+describe('Feedback Board Container ', () => {
+  it('can be rendered without content.', () => {
+    const wrapper = shallow(<FeedbackBoardContainer {...feedbackBoardContainerProps} />);
+    expect(wrapper.children().dive().html()).toBe(
+      '<div class="ms-Spinner initialization-spinner root-41"><div class="ms-Spinner-circle ms-Spinner--large circle-42">' +
+      '</div><div class="ms-Spinner-label label-43">Loading...</div></div>');
```

Comment:
Is there an async method for waiting and testing when the data has been loaded successfully?

Reply:
oh good question - so something we can do is use the `mount` command. The reason the data hasn't been loaded is that `shallow` doesn't really call the `componentDidMount` method which populates the data. TBH ... I had so many issues getting any test to work for this component. I think we would need to start mocking out all of the services in the `dal` directory in order to successfully call componentDidMount for these more complex components.

Reply:
Ah, ok, that makes sense, thank you!